motionplan/armplanning/real_test.go

```diff
@@ -142,3 +144,93 @@ func TestSandingLargeMove1(t *testing.T) {
+		{{180 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{150 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{120 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{90 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{60 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{30 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+		{{0 * 3.1415 / 180.0}, {0}, {-90 * 3.1415 / 180.0}, {0}, {0}, {0}},
+	}
+	// the only change here is in joint 0 in increments of 30, while all the other joints are kept at a constant value
+	// below is change in joint 0 in degrees:
+	// 0 -> 30 -> 60 -> 90 -> 120 -> 150 -> 180 -> 180 -> 150 -> 120 -> 90 -> 60 -> 30 -> 0
+
+	// determine pose given elements of idealJointValues
+	pifs := []*referenceframe.PoseInFrame{}
+	for _, pos := range idealJointValues {
+		pose, err := armKinematics.Transform(pos)
+		test.That(t, err, test.ShouldBeNil)
+		posInF := referenceframe.NewPoseInFrame(referenceframe.World, pose)
+		pifs = append(pifs, posInF)
+	}
+
+	// construct framesystem
+	fs := referenceframe.NewEmptyFrameSystem("pirouette")
+	err = fs.AddFrame(armKinematics, fs.World())
+	test.That(t, err, test.ShouldBeNil)
+
+	for iter := 0; iter < 10; iter++ {
+		// keep track of what the value of j0 previously was
+		prevJ0Value := 0.
+
+		// all we care about is the plan and not actually executing it
+		startState := NewPlanState(nil, map[string][]referenceframe.Input{armName: idealJointValues[0]})
+
+		// iterate through pifs and create a plan which gets the arm there
+		for i, p := range pifs {
+			t.Run(fmt.Sprintf("iteration: %d, pifs index: %d, pif: %v", iter, i, p), func(t *testing.T) {
+				subLogger := logging.NewTestLogger(t)
+				// construct req and get the plan
+				goalState := NewPlanState(map[string]*referenceframe.PoseInFrame{armName: p}, nil)
+				req := &PlanRequest{
+					FrameSystem: fs,
+					Goals:       []*PlanState{goalState},
+					StartState:  startState,
+				}
+				plan, err := PlanMotion(context.Background(), subLogger, req)
+				test.That(t, err, test.ShouldBeNil)
+
+				traj := plan.Trajectory()
+				// since we do not specify a constraint we are in the "free" motion profile which gives us a trajectory of length two
+				test.That(t, len(traj), test.ShouldEqual, 2) // ensure length is always two
+
+				// determine how much joint 0 has changed in degrees from this trajectory
+				allArmInputs, err := traj.GetFrameInputs(armName)
+				test.That(t, err, test.ShouldBeNil)
+				j0TrajStart := allArmInputs[0][0].Value
+				j0TrajEnd := allArmInputs[len(allArmInputs)-1][0].Value
+				j0Change := utils.RadToDeg(math.Abs(j0TrajEnd - j0TrajStart))
+
+				// figure out expected change given what the ideal change in joint 0 would be
+				idealJ0Value := idealJointValues[i][0].Value
+				expectedJ0Change := utils.RadToDeg(math.Abs(idealJ0Value-prevJ0Value)) + 1e-1 // add buffer
+
```

Comment:
can you log all the relevant things so that if there is an error we can see everything